File: PracticasFranc/CCola.js
class CCola {


    static a = new Array(10);
    static front = -1;
    static rear = -1;


    static isEmpty() {
        if (this.front === -1 || this.front > this.rear) {
            return true;
        } else {
            return false;
        }
    }

    static enqueue(x) {
        if (this.rear === this.a.length - 1) {
            console.log("La cola está llena");
        } else {
            if (this.front === -1) {
                this.front = 0;
            }
            this.rear = this.rear + 1;
            this.a[this.rear] = x;
        }
    }


    static dequeue() {
        if (this.isEmpty()) {
            console.log("La cola está vacía");
        } else {
            this.front = this.front + 1;
        }
    }

    // Imprimir la cola
    static imprimir() {
        if (this.isEmpty()) {
            console.log("La cola está vacía");
        } else {
            let salida = "";
            for (let i = this.front; i <= this.rear; i++) {
                salida += this.a[i] + " ";
            }
            console.log(salida.trim());
        }
    }


    static main() {

        this.enqueue(8);
        this.enqueue(5);
        this.enqueue(2);
        this.enqueue(3);
        this.dequeue();
        this.dequeue();
        this.imprimir();

        console.log("ID del procesador: " + (process.env.PROCESSOR_IDENTIFIER || "No disponible"));
        console.log("Arquitectura del procesador: " + (process.env.PROCESSOR_ARCHITECTURE || process.arch));
        console.log("Número de procesadores: " + require("os").cpus().length);
        console.log("Directorio temporal: " + require("os").tmpdir());
    }
}

CCola.main();

File: PracticasFranc/CCola.java
public class CCola {

    // Atributos de la cola
    static int a[] = new int[10];
    static int front = -1;
    static int rear = -1;

    // Método para verificar si la cola está vacía
    public static boolean isEmpty() {
        if (front == -1 || front > rear) {
            return true;
        } else {
            return false;
        }
    }

    // Método para insertar (enqueue)
    public static void enqueue(int x) {
        if (rear == a.length - 1) {
            System.out.println("La cola está llena");
        } else {
            if (front == -1) {
                front = 0;
            }
            rear = rear + 1;
            a[rear] = x;
        }
    }

    // Método para eliminar (dequeue)
    public static void dequeue() {
        if (isEmpty()) {
            System.out.println("La cola está vacía");
        } else {
            front = front + 1;
        }
    }

    // Método para imprimir la cola
    public static void imprimir() {
        if (isEmpty()) {
            System.out.println("La cola está vacía");
        } else {
            for (int i = front; i <= rear; i++) {
                System.out.print(a[i] + " ");
            }
            System.out.println();
        }
    }

    // Método principal (DEBE LUCIR EXACTAMENTE ASÍ)
    public static void main(String[] args) {

        enqueue(8);
        enqueue(5);
        enqueue(2);
        enqueue(3);
        dequeue();
        dequeue();
        imprimir();

        System.out.println("ID del procesador: " + System.getenv("PROCESSOR_IDENTIFIER"));
        System.out.println("Arquitectura del procesador: " + System.getenv("PROCESSOR_ARCHITECTURE"));
        System.out.println("Número de procesadores: " + System.getenv("NUMBER_OF_PROCESSORS"));
        System.out.println("Directorio temporal: " + System.getenv("TMP"));
    }
}
    